rules/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector.php

```diff
@@ -106,19 +102,11 @@ public function refactor(Node $node): ?Node
                 continue;
             }
 
-            if (! $classReflection instanceof ClassReflection) {
-                $classReflection = $this->reflectionResolver->resolveClassReflection($node);
-            }
-
-            // ClassReflection not detected, early skip
-            if (! $classReflection instanceof ClassReflection) {
-                return null;
-            }
-
-            $type = $this->allAssignNodePropertyTypeInferer->inferProperty(
+            $propertyName = (string) $this->getName($property);
+            $type = $this->assignToPropertyTypeInferer->inferPropertyInClassLike(
```

Comment:
Thanks 👍 

The `assignToPropertyTypeInferer` should be removed as well, to avoid checking all the possible types and assigns.
This rule should be as simple as possible.

Reply:
if no check on the assignments on other methods, then, reset on other method will cause error, as it can't detect nullability, this ensure nullability or reset by other type applied,eg:

```php
protected function setUp()
{
   $this->a = $this->createMock(Foo::class);
}

protected function reset()
{
    $this->a = null;
}
```

if we check only on `setUp()`, that will cause it only detect as `MockObject`, which wrong

Reply:
the only apply `MockObject` or `?MockObject` verify already checked on later line 121 on this PR

Reply:
I added test for skip with non-mock object type, eg: skip string https://github.com/rectorphp/rector-src/pull/6181/commits/298f24c7e2bdf23b9ddc208066ba68fd6fb2e409

Reply:
I see, let's go with this version for now then.